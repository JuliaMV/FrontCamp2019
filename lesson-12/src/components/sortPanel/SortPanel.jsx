import React from 'react';
import PropTypes from 'prop-types';

import Filter from 'components/filter/Filter';
import css from './SortPanel.module.css';

const SortPanel = (props) => {
  const {
    description, isFilter, activeFilter, filterHandler,
  } = props;
  let filter = null;
  if (isFilter) {
    filter = <Filter title="sort by" fields={['release date', 'rating']} activeFilter={activeFilter} filterHandler={filterHandler} />;
  }

  return (
    <div className={css.wrapper}>
      <div className={css.container}>
        <p className={css.description}>{description}</p>
        {filter}
      </div>
    </div>
  );
};


SortPanel.propTypes = {
  description: PropTypes.string,
  isFilter: PropTypes.bool,
  activeFilter: PropTypes.string,
  filterHandler: PropTypes.func,
};

SortPanel.defaultProps = {
  description: 'Sort by',
  isFilter: false,
  activeFilter: '',
  filterHandler: () => {},
};

export default SortPanel;
