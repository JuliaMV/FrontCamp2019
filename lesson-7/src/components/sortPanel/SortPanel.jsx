import React from 'react';

import Filter from '../filter/Filter';
import css from './SortPanel.module.css';

const SortPanel = () => (
  <div className={css.container}>
    <p className={css.description}>7 movie found</p>
    <Filter title="sort by" fields={['release date', 'rating']} />
  </div>
);

export default SortPanel;
