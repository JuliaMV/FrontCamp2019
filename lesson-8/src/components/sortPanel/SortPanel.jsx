import React from 'react';
import PropTypes from 'prop-types';

import Filter from 'components/filter/Filter';
import css from './SortPanel.module.css';

const SortPanel = (props) => {
  const {
    description, isFilter, activeFilter, filterHandler,
  } = props;
  let filter = null;
  if (isFilter) {
    filter = <Filter title="sort by" fields={['release date', 'rating']} activeFilter={activeFilter} filterHandler={filterHandler} />;
  }
  console.log(props);
  return (
    <div className={css.wrapper}>
      <div className={css.container}>
        <p className={css.description}>{description}</p>
        {filter}
      </div>
    </div>
  );
};


SortPanel.propTypes = {
  description: PropTypes.string.isRequired,
  isFilter: PropTypes.bool,
  activeFilter: PropTypes.string,
  filterHandler: PropTypes.func,
};

SortPanel.defaultProps = {
  isFilter: false,
  activeFilter: '',
  filterHandler: () => {},
};

export default SortPanel;
