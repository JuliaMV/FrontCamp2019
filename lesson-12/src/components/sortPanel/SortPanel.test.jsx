import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import SortPanel from './SortPanel';

Enzyme.configure({ adapter: new Adapter() });


describe('SortPanel component', () => {
  test('should render with default props', () => {
    const component = shallow(<SortPanel />);
    expect(toJson(component)).toMatchSnapshot();
  });

  test('should render with enabled filter', () => {
    const component = shallow(<SortPanel isFilter />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
