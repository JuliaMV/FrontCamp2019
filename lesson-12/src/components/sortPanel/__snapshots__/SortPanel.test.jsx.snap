// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortPanel component should render with default props 1`] = `
<div>
  <div>
    <p>
      Sort by
    </p>
  </div>
</div>
`;

exports[`SortPanel component should render with enabled filter 1`] = `
<div>
  <div>
    <p>
      Sort by
    </p>
    <Filter
      activeFilter=""
      fields={
        Array [
          "release date",
          "rating",
        ]
      }
      filterHandler={[Function]}
      title="sort by"
    />
  </div>
</div>
`;
